keps/sig-node/4210-max-image-gc-age/README.md

```diff
@@ -0,0 +1,395 @@
+the primary goal of garbage collection is to ensure images don't clutter the disk too much and cause it to fill up needlessly.
+However, having the Kubelet be purely reactive means that images will clutter the disk and cause it to fill up. While there aren't reported cases
+where this causes issues, it is inefficient with disk space and can cause the Kubelet to scramble to save disk space when the threshold is met.
+
+An additional approach is to define a way for an admin to request images are cleaned up after they're unused for a certain period of time.
+This would reduce the frequency of the disk usage hitting the level, and provide an admin more flexibility in how garbage collection is defined.
+
+The proposal of this KEP is to add an option to the KubeletConfiguration object that looks like:
+```
+	// ImageMaximumGCAge is the maximum age an image can be unused before it is garbage collected.
+    // The default of this field is 0, which disables it.
+    // +optional
+	ImageMaximumGCAge metav1.Duration
+```
+
+To begin, this option will be set to 0, which will be interpreted as "disabled". In the future, a more reasonable default may be chosen.
+
+This option will only be adhered to if the feature gate ImageMaximumGCAge is configured for the Kubelet.
+
+### User Stories (Optional)
+
+#### Story 1
+
+- As a cluster admin, I would like my unused images to be garbage collected in a timely manner, and not occupy disk space forever.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+- If set incorrectly the ImageMaximumGCAge option could cause unneeded image pulls. For instance, if a Cron job ran
+once a week, but the ImageMaximumGCAge was set to less than a week, that image would get pulled every week, causing needless
+traffic from the registry
+    - Proper documentation on this is the best way to mitigate this risk.
+    - Defining a good default for this value will be similarly tricky.
+- Reliability of image age
+    - Good testing will mitigate/fix any errors
+- New, undiscovered races
+    - If the max image gc age is set very low, will the kubelet race with itself and remove the image right after pulling it?
+    - May need to define a minimum maximum gc age to prevent races like this.
+- Runtime misbehavior
+    - It's possible the runtime won't GC the image and kubelet will begin thrashing on the image.
+    - Runtime maintainers should ensure to avoid this situation
+    
+
+## Design Details
+
+Add an option to the Kubelet configuration:
+```
+	// ImageMaximumGCAge is the maximum age an image can be unused before it is garbage collected.
+    // The default of this field is 0, which disables it.
```

Comment:
so `0` disables it. So if we want to implement the behavior of "no cache" - we will need to ask to set a threshold of 1 second or something?

Reply:
yeah I had originally thought to set the default to int64 max, but it was suggested 0 may be better. I think functionally 1s is no cache, especially if we're only garbage collecting on the heartbeat of 5 minutes